### PR TITLE
Fixed copy internal-link into new language

### LINKED ISSUE #2313: Page translation for page with type "internal link"
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.2.0
| Browser Version | All

#### Actual Behavior
Adding a translation for a page with "internal  link" as page type does not work as expected:

When selecting "Create a new empty page in this language": 
- A page with type internal link is created
- The content of "Linked content" is copied and "No Title (/main/sub-1)" is shown (even if the linked page is not available in this language)
- Selecting "content" as page type and saving afterwords leads to an error ("Document has no title (title is required for auto name behavior)")

internal-link

When selecting "Create a new page in this language and copy content from": 
- The content of "Linked content" is copied and "No Title (/main/sub-1)" is shown (even if the linked page is not available in this language)

#### Expected Behavior
When selecting "Create a new empty page in this language": 
- A page with type content should be created.

When selecting "Create a new page in this language and copy content from": 
- The content of "Linked content" should not be copied if the selected page is not available in the current language.

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #2313
| Related issues/PRs | https://github.com/sulu/sulu-document-manager/pull/88 and https://github.com/massiveart/husky/pull/652
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR updates sulu-document-manager and additionally fixes some javascript issues with copy internal link content to a new language.

#### Why?

There were several issues with copy internal-link pages.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@ CHANGELOG for Sulu
 ==================
 
 * dev-develop
+    * HOTFIX      #2367 [ContentBundle]       Fixed copy internal-link into new language
```
Comment (danrot) [inline CHANGELOG.md]: Not a hotfix anymore.

**Comment by danrot** [comment]
Please add the new methods from the `LocaleBehavior` to the `UPGRADE.md`.

File: src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspector.php [hunk 2]
```diff
@@ -110,9 +111,17 @@ public function getContentPath(PathBehavior $document)
      */
     public function getStructureMetadata(StructureBehavior $document)
     {
+        $alias = $this->getMetadata($document)->getAlias();
+        $structureType = $document->getStructureType();
+        if (null === $structureType) {
```
Comment (danrot) [inline src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspector.php]: Somehow strange that this is necessary...

File: src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspector.php [hunk 3]
```diff
@@ -110,9 +111,17 @@ public function getContentPath(PathBehavior $document)
...
+            $webspace = $this->webspaceManager->findWebspaceByKey($this->getWebspace($document));
+            $structureType = $webspace->getTheme()->getDefaultTemplate($alias);
+            $document->setStructureType($structureType);
```
Comment (danrot) [inline src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspector.php]: For me that's also unexpected behavior. Just asking for the metadata should not automatically set it.

**Comment by danrot** [comment]
One test is failing, related to the comment I added before.

What I have tests works well so far, this includes:

- The problem described in #2313 
- Having ghost pages in the column tree
- Having shadows in the column tree
- Set a shadow and access it on the website

However, it is a bit hard to test, because of the published bug mentioned today morning...

**Comment by wachterjohannes** [comment]
@danrot i have done this now in another way.

File: src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspector.php [hunk 4]
```diff
@@ -112,7 +113,23 @@ public function getStructureMetadata(StructureBehavior $document)
...
+    }
+
+    /**
+     * Return the default structure for the given StructureBehavior implementing document.
+     *
+     * @param StructureBehavior $document
+     *
+     * @return StructureMetadata
+     */
+    public function getDefaultStructureType(StructureBehavior $document)
```
Comment (danrot) [inline src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspector.php]: This method doesn't return anything really related to the document, so I would prefer to not have it in the `DocumentInspector`. Also, from what I've seen this method is only used in one place, why not just putting this code there?

**Comment by danrot** [comment]
I only created a page in english, and the title does not appear, it is shown as if it would be a ghost page:
image

This can also be confirmed when looking at the request, which is sent to http://sulu-test.lo/admin/api/nodes/d4c904fa-4f00-4875-847b-d2741633f5de?webspace=sulu_io&language=en&complete=false&ghost-content=true:

```json
{
  "url": "\/test",
  "title": "Test",
  "navContexts": [
    
  ],
  "nodeType": 1,
  "nodeState": 2,
  "published": "2016-05-23T08:00:33+0200",
  "id": "d4c904fa-4f00-4875-847b-d2741633f5de",
  "order": 20,
  "_permissions": {
    "view": true,
    "add": true,
    "edit": true,
    "delete": true,
    "archive": true,
    "live": true,
    "security": true
  },
  "template": "default",
  "originTemplate": "default",
  "internal": false,
  "localizedTemplate": "Default",
  "path": "\/test",
  "concreteLanguages": [
    "en"
  ],
  "type": {
    "name": "ghost",
    "value": "en"
  },
  "enabledShadowLanguages": [
    
  ],
  "linked": null,
  "publishedState": true,
  "hasSub": false,
  "_hash": "a63851d3a71306147834b16f4f73ad26"
}
```

There is a `type` attribute saying the current page is a ghost. This should not be the case, and it looks like there is no test existing covering that, can you maybe add one?

**Comment by danrot** [comment]
@wachterjohannes Tests are failing.

**Pull request merged**